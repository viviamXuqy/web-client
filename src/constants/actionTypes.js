// user
export const FETCH_USER = 'FETCH_USER';
export const SIGNIN = 'SIGNIN'; // login
export const UPDATE_USER = 'UPDATE_USER';

// image
export const GET_IMG = 'GET_IMG'; // server端通用获取图片

// results 识别结果
export const FETCH_RESULTS = 'FETCH_RESULTS'; // 识别结果list
export const FETCH_RESULT = 'FETCH_RESULT'; // 识别结果详情
export const SUBMIT_RESULT = 'SUBMIT_RESULT';
export const DELETE_RESULT = 'DELETE_RESULT'; // 删除识别结果
export const UPLOAD_RESULT_FEATURE = 'UPLOAD_RESULT_FEATURE'; // 以图搜图: 上传图片识别出结构化信息
export const DELETE_RESULT_ALL = 'DELETE_RESULT_ALL'; // 删除全部识别结果

// tasks 任务结果
export const FETCH_TASKS = 'FETCH_TASKS'; // 任务列表
export const SUBMIT_TASK = 'SUBMIT_TASK'; // 添加任务
export const DELETE_TASK = 'DELETE_TASK'; // 删除任务
export const UPDATE_TASK_STATUS = 'UPDATE_TASK_STATUS'; // 开关状态
export const UPDATE_TASK_STATUS_LOT = 'UPDATE_TASK_STATUS_LOT'; // 批量开关状态
export const DELETE_TASK_ALL = 'DELETE_TASK_ALL'; // 删除全部任务
export const SET_TASK = 'SET_TASK'; // 任务设置
export const GET_TASK_NUM = 'GET_TASK_NUM'; // 实时(开启/总数）

// 任务总结果列表
export const FETCH_TASKRESULTS = 'FETCH_TASKRESULTS'; // 任务总结果列表list
export const DELETE_TASKRESULTS = 'DELETE_TASKRESULTS'; // 删除任务总结果

// cameras 设备管理
export const FETCH_CAMERAS = 'FETCH_CAMERAS'; // 设备list
export const FETCH_CAMERA = 'FETCH_CAMERA'; // 设备Camera
export const ADD_CAMERA = 'ADD_CAMERA'; // 添加Camera
export const DELETE_CAMERA = 'DELETE_CAMERA'; // 删除设备
export const UPDATE_CAMERA_STATUS = 'UPDATE_CAMERA_STATUS';
export const UPDATE_CAMERA_AREA = 'UPDATE_CAMERA_AREA'; // 更新设置摄像头区域
export const UPDATE_CAMERA_REC = 'UPDATE_CAMERA_REC'; // 摄像头监控开关
export const UPDATE_CAMERA_RECS = 'UPDATE_CAMERA_RECS'; // 摄像头监控开关s
export const GET_CAMERA_IMG = 'GET_CAMERA_IMG'; // 摄像头显示图
export const GET_CAMERA_NUM = 'GET_CAMERA_NUM'; // 摄像头(开启/总数）
export const RENAME_CAMERA = 'RENAME_CAMERA'; // 重命名摄像头
export const UPLOAD_EXCEL_FILE = 'UPLOAD_EXCEL_FILE'; // 上传批量刷新摄像头列表
export const GET_CAMERA_TASK_NUM = 'GET_CAMERA_TASK_NUM'; // 获取到摄像头关联的任务
// Nvr
export const FETCH_NVRS = 'FETCH_NVRS'; // NVRlist
export const ADD_NVR = 'ADD_NVR'; // 添加NVR
export const MODIFY_NVR = 'MODIFY_NVR'; // 修改NVR
export const DELETE_NVR = 'DELETE_NVR'; // 删除NVR

// 卡口
export const FETCH_BAYONETS = 'FETCH_BAYONETS'; // 卡口列表
export const FETCH_BAYONET_CAMERAS = 'FETCH_BAYONET_CAMERAS'; // 卡口下摄像头列表
export const FETCH_BAYONET_SDK_CAMERAS = 'FETCH_BAYONET_SDK_CAMERAS'; // 卡口下SDK摄像头列表
export const SUBMIT_BAYONET = 'SUBMIT_BAYONET'; // 添加卡口
export const SUBMIT_BAYONET_SDK = 'SUBMIT_BAYONET_SDK'; // 添加卡口独立
export const DELETE_BAYONET = 'DELETE_BAYONET'; // 删除卡口
export const RESET_BAYONET_NAME = 'RESET_BAYONET_NAME'; // 重命名卡口

// 视频/图片源
export const FETCH_RESOURCES = 'FETCH_RESOURCES'; // 源列表
export const FETCH_RESOURCE_TREE = 'FETCH_RESOURCE_TREE'; // 源下树列表
export const FETCH_RESOURCE_TREE_FILE = 'FETCH_RESOURCE_TREE_FILE'; // 源下文件列表
export const UPLOAD_FILE = 'UPLOAD_FILE'; // 上传文件

// 结构化数据
export const FETCH_FEATURES = 'FETCH_FEATURES'; // 拿到所有结构化数据

// filter
export const UPDATE_FILTER = 'UPDATE_FILTER'; // 检索条件过滤


// clear data
export const CLEAR_DATAS = 'CLEAR_DATAS'; // 清除store中相应的数据

// error message
export const SHOW_ERROR_MESSAGE = 'SHOW_ERROR_MESSAGE';
export const HIDE_ERROR_MESSAGE = 'HIDE_ERROR_MESSAGE';

// loading
export const SHOW_LOADING = 'SHOW_LOADING';
export const HIDE_LOADING = 'HIDE_LOADING';

// 系统设置
export const FETCH_SDKS = 'FETCH_SDKS'; // sdk传输协议列表
export const ADD_SDK = 'ADD_SDK'; // 添加协议
export const DELETE_SDK = 'DELETE_SDK'; // 删除任务
export const UPDATE_SDK_STATUS = 'UPDATE_SDK_STATUS'; // 开关状态
export const MODIFY_SDK = 'MODIFY_SDK'; // 修改协议
export const MODIFY_WORK_MODE = 'MODIFY_WORK_MODE'; // 修改工作配置
export const RECOVER_CONFIG = 'RECOVER_CONFIG'; // 恢复系统配置
export const SAVE_NTP = 'SAVE_NTP'; // 保存Ntp校时
export const SAVE_MANUAL = 'SAVE_MANUAL'; // 保存手动校时
export const FETCH_LOGS = 'FETCH_LOGS'; // 日志列表
export const DELETE_LOG = 'DELETE_LOG'; // 删除日志
export const FETCH_USERS = 'FETCH_USERS'; // 用户列表
export const ADD_USER = 'ADD_USER'; // 添加用户
export const DELETE_USER = 'DELETE_USER'; // 删除用户
export const UPDATE_USER_STATUS = 'UPDATE_USER_STATUS'; // 用户启用/停用状态
export const MODIFY_USER = 'MODIFY_USER'; // 修改用户

// 智能应用-机动车检测
export const FETCH_VEHICLE_TASKS = 'FETCH_VEHICLE_TASKS'; // 机动车任务列表
export const FETCH_ACTIVE_TASKS = 'FETCH_ACTIVE_TASKS'; // 进行中的任务列表
export const ADD_VEHICLE_TASK = 'ADD_VEHICLE_TASK'; // 创建机动车任务
export const CLEAR_VEHICLE_TASK = 'CLEAR_VEHICLE_TASK'; // 清空任务
export const DELETE_VEHICLE_TASK = 'DELETE_VEHICLE_TASK'; // 批量删除任务
export const CLOSE_VEHICLE_TASK = 'CLOSE_VEHICLE_TASK'; // 批量关闭任务
export const GET_VEHICLE_RESULT = 'GET_VEHICLE_RESULT'; // 获取机动车任务结果详情
export const DELETE_VEHICLE_RESULT = 'DELETE_VEHICLE_RESULT'; // 批量删除结果
export const CLEAR_VEHICLE_RESULT = 'CLEAR_VEHICLE_RESULT'; // 清空结果
export const START_VEHICLE_TASK = 'START_VEHICLE_TASK'; // 开启任务
export const FETCH_VEHICLE_RESULTS = 'FETCH_VEHICLE_RESULTS'; // 获取机动车驾驶监控-结果列表
export const FETCH_MONITOR_PLAYPATH = 'FETCH_MONITOR_PLAYPATH'; // 获取播放流地址
export const FETCH_MONITOR_ANALYSIS = 'FETCH_MONITOR_ANALYSIS'; // 获取流量统计数据
export const FETCH_MONITOR_DATA = 'FETCH_MONITOR_DATA'; // 获取实时机动车, 行人列表
export const START_MONITOR = 'START_MONITOR'; // 实时交通监控开启
export const STOP_MONITOR = 'STOP_MONITOR'; // 实时交通监控关闭

// 智能应用-违章监控
export const FETCH_VIOLATIONS = 'FETCH_VIOLATIONS'; // 违章任务列表
export const DELETE_VIOLATION_TASKS = 'DELETE_VIOLATION_TASKS'; // 批量删除任务
export const CLEAR_VIOLATION_TASKS = 'CLEAR_VIOLATION_TASKS'; // 清空任务
export const SWITCH_VIOLATION_TASKS = 'SWITCH_VIOLATION_TASKS'; // 违章监控开关
export const ADD_VIOLATION_TASK = 'ADD_VIOLATION_TASK'; // 新增违章监控任务
export const MODIFY_VIOLATION_TASKS = 'MODIFY_VIOLATION_TASKS'; // 修改违章任务
export const FETCH_VIOLATION_RESULTS = 'FETCH_VIOLATION_RESULTS'; // 违章结果列表
export const DELETE_VIOLATION_RESULTS = 'DELETE_VIOLATION_RESULTS'; // 删除违章结果
export const CLEAR_VIOLATION_RESULTS = 'CLEAR_VIOLATION_RESULTS'; // 清空违章结果
export const GET_VIOLATION_RESULT = 'GET_VIOLATION_RESULT'; // 违章结果详细信息-城市道路违停

export const RESET_MONITOR_DATA = 'RESET_MONITOR_DATA'; // 重置数据
export const FETCH_CAMERA_TASKS = 'FETCH_CAMERA_TASKS'; // 已有摄像头名称列表

// 智能应用-复省车牌
export const GET_REVIEW_PLATE_TASKS = 'GET_REVIEW_PLATE_TASKS';
export const GET_REVIEW_PLATE_TASK = 'GET_REVIEW_PLATE_TASK';
export const GET_REVIEW_PLATE_PREVIEW_IMAGE = 'GET_REVIEW_PLATE_PREVIEW_IMAGE';
export const PUT_REVIEW_PLATE_AREA = 'PUT_REVIEW_PLATE_AREA';
export const PUT_REVIEW_PLATE_TASK_START = 'PUT_REVIEW_PLATE_TASK_START';
export const GET_REVIEW_PLATE_RESULTS = 'GET_REVIEW_PLATE_RESULTS';
export const GET_REVIEW_PLATE_TASK_STATISTICS = 'GET_REVIEW_PLATE_TASK_STATISTICS';
