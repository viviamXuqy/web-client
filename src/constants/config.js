export const isDev = true;//process.env.NODE_ENV === 'development';
export const isProd = process.env.NODE_ENV === 'production';
export const proxy = 'http://localhost:3001/api';

const CONFIG = {
  ip: window.api().url,
  port: window.api().port,
  useLocation: window.useLocation(),
  lang: window.defaultLang(),
};

export const DEFALUT_LANG = CONFIG.lang;

export const API = CONFIG.useLocation ? `${window.location.protocol}//${window.location.hostname}:${CONFIG.port}` : `${CONFIG.ip}:${CONFIG.port}`;

export const API_URLS = {
  dev: '/api',
  prod: `${API}/api`,
};

export const API_URL =
  (isProd && API_URLS.prod) ||
  (isDev && API_URLS.dev);

export const BASE64_IMG_JPEG = 'data:image/jpeg;base64,';
export const TASK_TYPE = {
  ALL: 0, ANALYSIS: 1, VIDEO: 2, PIC: 3,
}; // 任务类型（1实时，2视频，3图集）
