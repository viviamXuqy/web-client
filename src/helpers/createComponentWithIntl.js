import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line
import LocaleProvider from 'containers/Locale';
import { setStore } from '../utils/storage';
import { nodeWithIntlProp } from './intl-enzyme-test-helper';

const createComponentWithIntl = (children, props = { lang: 'zh-CN' }) => {
  setStore('locale', props.lang);
  return renderer.create(
    <LocaleProvider {...props}>
      {nodeWithIntlProp(children)}
    </LocaleProvider>,
  );
};

export default createComponentWithIntl;
