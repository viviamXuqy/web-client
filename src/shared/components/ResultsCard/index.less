.results-card-label {
  color: rgb(137, 137, 137);
  font-size: 12px;
}

.results-card-label-content {
  color: rgb(28, 28, 31);
  font-size: 12px;
}

.results-card-div {
  padding: 10px;
  width: 100%;
  height: 100%;
}

.results-card-div-border {
  padding: 8px;
  width: 100%;
  height: 100%;
  border: 2px solid;
  border-color: rgba(66, 109, 160, 1);
}

.results-card-content {
  .ant-btn-dashed {
    border: 0;
    background-color: transparent;
    outline: none;

    &:hover, &:focus {
      border: 0;
      background-color: transparent;
      outline: none;
    }
  }

  .large-icon {
    font-size: 18px;
  }

  .divhand {
    cursor: pointer;
  }
}

.results-card-content__btnGroups {
  float: right;

  .ant-radio-wrapper {
    margin-right: 0;
  }
}

.results-card__lazyload {
  opacity: 1;
  animation: results-card__lazyload-show 2s ease-in-out 1;
}

@keyframes results-card__lazyload-show {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}
