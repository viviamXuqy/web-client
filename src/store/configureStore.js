import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import api from '../middleware/api';
import error from '../middleware/error';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const isDev = process.env.NODE_ENV === 'development';
const loggerMiddleware = isDev ? [createLogger()] : [];

const configureStore = history => {
  const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(
      routerMiddleware(history),
      thunk, api, error, ...loggerMiddleware,
    )),
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../reducers', () => {
      const nextRootReducer = require('../reducers').default // eslint-disable-line
      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
};

export default configureStore;
