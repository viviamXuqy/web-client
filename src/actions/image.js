import { CALL_API } from '../middleware/api';
import { GET_IMG } from '../constants/actionTypes';

// server端通用获取图片
export const getImg = (id, scale, quality) => ({ // eslint-disable-line
  [CALL_API]: {
    type: GET_IMG,
    method: 'GET',
    endpoint: `/image/${id}/${scale === undefined ? 10 : scale}/${quality === undefined ? 100 : quality}`,
    suppressLoading: true,
    suppressError: true,
  },
});
