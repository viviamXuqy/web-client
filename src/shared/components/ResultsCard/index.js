import React from 'react';
import { object, func, bool } from 'prop-types';
import { Avatar, Radio, Button, Icon, Spin } from 'antd';
import { FormattedMessage } from 'react-intl';

import moment from 'moment';
import Cookies from 'js-cookie';

import { BASE64_IMG_JPEG } from '../../../constants/config';
import { decodeUnicode, compressImage } from '../../../utils';

import './index.less';

const _isMountedObjArr = [];

class ResultsCard extends React.Component {
  static propTypes = {
    isResultList: bool.isRequired, // 是否是识别结果列表
    result: object.isRequired,
    fetchResult: func.isRequired,
    getImg: func.isRequired,
    onDelete: func,
    onSelect: func,
    showResultDetail: func,
    isSelected: bool,
  }

  static defaultProps = {
    onDelete: () => {},
    onSelect: () => {},
    showResultDetail: () => {},
    isSelected: false,
  }

  state = {
    imgUrl: '',
    resultId: '',
    imgLoading: true, // 图像的loading
  }

  componentDidMount() {
    if (this.props.result && _isMountedObjArr[this.props.result.id]) {
      this.handleImgShow();
    }
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.result && nextProps.result.id !== prevState.resultId) {
      _isMountedObjArr[nextProps.result.id] = true;
      return {
        imgLoading: true,
        imgUrl: '',
        resultId: nextProps.result.id,
      };
    }
    return null;
  }

  componentDidUpdate() {
    if (this.props.result && _isMountedObjArr[this.props.result.id]) {
      this.handleImgShow();
    }
  }

  componentWillUnmount() {
    const { result } = this.props;
    _isMountedObjArr[result.id] = false;
  }

  handleImgShow = () => {
    const { id: resultId } = this.props.result;
    const self = this;

    const { getImg, result } = this.props;
    if (result.image) {
      compressImage(result.image, 693, resultId).then(({ src, data }) => {
        if (_isMountedObjArr[data]) {
          _isMountedObjArr[data] = false;
          self.setState({
            imgUrl: src,
            imgLoading: false,
          });
        }
      });
    } else {
      getImg(resultId, 2, 10).then(({ response }) => {
        if (response) {
          if (_isMountedObjArr[response.id]) {
            _isMountedObjArr[response.id] = false;
            self.setState({
              imgLoading: false,
              imgUrl: BASE64_IMG_JPEG + response.image,
            });
          }
        }
      });
    }
  }

  handleImgClick = () => {
    this.handleMouseDown();
  }

  handleMouseDown = event => {
    const {
      result, showResultDetail, fetchResult, onDelete, onSelect,
    } = this.props;

    if (event && event.target.name === 'resultDel') {
      onDelete([result.id]);
    } else if (event && event.target.name === 'resultSelect') {
      onSelect(result.id);
    } else {
      fetchResult(result.id)
        .then(({ response }) => {
          if (response) {
            showResultDetail({
              ...response,
            });
          }
        });
    }
  }

  render() {
    const { result, isResultList, isSelected } = this.props;
    const { imgUrl, imgLoading } = this.state;
    const user = JSON.parse(Cookies.get('user'));
    const element = (
      <div className="results-card-div">
        <Spin spinning={imgLoading}>
          <Avatar
            className="results-card__lazyload"
            shape="square"
            size="large"
            src={imgUrl}
            onClick={() => this.handleImgClick()}
          />
        </Spin>
        <div className="results-card-content">
          <div className="divhand" onMouseDown={event => this.handleMouseDown(event)}>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.results.resultId" />: </span>
              <span className="results-card-label-content">{result.id && result.id.length > 13 ? `${result.id.slice(0, 13)}...` : result.id}</span>
            </p>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.results.taskId" />: </span>
              <span className="results-card-label-content">{result.taskId && result.taskId.length > 13 ? `${result.taskId.slice(0, 13)}...` : result.taskId}</span>
            </p>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.public.time" />: </span>
              <span className="results-card-label-content">{result.time ? moment(Number(result.time)).format('YYYY-MM-DD HH:mm:ss') : <FormattedMessage id="App.public.no" />}</span>
            </p>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.public.type" />: </span>
              <span className="results-card-label-content">{decodeUnicode(result.targetType)}</span>
            </p>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.results.taskType" />: </span>
              <span className="results-card-label-content"><FormattedMessage id={`App.results.${result.taskType}`} /></span>
            </p>
          </div>
        </div>
      </div>
    );

    const elementResult = (
      <div
        className={isSelected ? 'results-card-div-border' : 'results-card-div'}
      >
        <Spin spinning={imgLoading}>
          <Avatar
            className="results-card__lazyload"
            shape="square"
            size="large"
            src={imgUrl}
            onClick={() => this.handleImgClick()}
          />
        </Spin>
        <div
          className="results-card-content"
        >
          <div className="divhand" onMouseDown={event => this.handleMouseDown(event)}>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.results.resultId" />: </span>
              <span className="results-card-label-content">{result.id && result.id.length > 21 ? `${result.id.slice(0, 21)}...` : result.id}</span>
            </p>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.results.taskId" />: </span>
              <span className="results-card-label-content">{result.taskId && result.taskId.length > 21 ? `${result.taskId.slice(0, 21)}...` : result.taskId}</span>
            </p>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.public.time" />: </span>
              <span className="results-card-label-content">{result.time ? moment(Number(result.time)).format('YYYY-MM-DD HH:mm:ss') : <FormattedMessage id="App.public.no" />}</span>
            </p>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.public.type" />: </span>
              <span className="results-card-label-content">{decodeUnicode(result.targetType)}</span>
            </p>
            <p>
              <span className="results-card-label"><FormattedMessage id="App.results.taskType" />: </span>
              <span className="results-card-label-content"><FormattedMessage id={`App.results.${result.taskType}`} /></span>
              <span className="results-card-content__btnGroups">
                <Button
                  name="resultDel"
                  size="small"
                  type="dashed"
                  disabled={user.grade === '3'}
                >
                  <Icon className="large-icon" type="delete" />
                </Button>
                <Radio name="resultSelect" checked={isSelected} />
              </span>
            </p>
          </div>
        </div>
      </div>
    );

    return (
      <div>
        {!isResultList && element}
        {isResultList && elementResult}
      </div>
    );
  }
}

export default ResultsCard;
