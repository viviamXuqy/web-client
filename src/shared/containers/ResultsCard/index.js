import { connect } from 'react-redux';

import ResultsCard from '../../../shared/components/ResultsCard';
import { fetchResult } from '../../../actions/results';
import { getImg } from '../../../actions/image';

export default connect(null, {
  fetchResult,
  getImg,
})(ResultsCard);
