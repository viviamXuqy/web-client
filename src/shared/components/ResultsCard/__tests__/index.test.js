import React from 'react';
import createComponentWithIntl from 'helpers/createComponentWithIntl';
import { mountWithIntl } from 'helpers/intl-enzyme-test-helper';

import ResultsCard from 'shared/components/ResultsCard';

import Cookies from 'js-cookie';

import { BASE64_IMG_JPEG } from 'constants/config';

import { img } from '../../../../../api/schema/image';

import configureStore from '../../../../store/__tests__/configureStore.test';

const { state } = configureStore;

function initializeData() {
  Cookies.set('user', JSON.stringify(state.user));
}

const fetchResult = jest.fn(() => Promise.resolve({ response: 'success' }));
const onDelete = jest.fn();
const onSelect = jest.fn();
const showResultDetail = jest.fn();

const result = {
  id: 1,
  taskId: 1,
  time: '1529355966000',
  taskType: 1,
  targetType: 'car',
  taskName: 'test1',
  address: 'test2',
  ratio: '3392x2008',
  image: '',
};
const getImg = jest.fn(id => Promise.resolve({ response: { image: img, id } }));

const initProps = {
  isResultList: true,
  result,
  fetchResult,
  getImg,
  onDelete,
  onSelect,
  showResultDetail,
  isSelected: false,
};

const setup = (props = initProps) => {
  const component = <ResultsCard {...props} />;
  const componentWithProvider = createComponentWithIntl(component);
  const wrapper = mountWithIntl(component);

  return {
    component,
    componentWithProvider,
    wrapper,
  };
};

describe('ResultsCard component', () => {
  beforeEach(() => {
    initializeData();
  });

  it('should render without errors', () => {
    const { wrapper } = setup();

    expect(wrapper).toBeTruthy();
  });

  it("should render correct things when 'isResultList' is false", () => {
    const { wrapper } = setup();

    expect(wrapper.find('Radio').length).toBe(1);

    wrapper.setProps({
      ...wrapper.props(),
      isResultList: false,
    });

    expect(wrapper.find('Radio').length).toBe(0);
  });

  it('should render correct things', () => {
    const { wrapper } = setup();

    expect(wrapper.find('Radio').length).toBe(1);

    expect(wrapper.find('.results-card-content').length).toBe(1);
  });

  it("should render right thing and don't change unexpected", () => {
    const { componentWithProvider } = setup();
    const tree = componentWithProvider.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should call 'componentWillUnmount' correct when unmount", () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();

    const spyWillUnmount = jest.spyOn(instance, 'componentWillUnmount');

    expect(spyWillUnmount.mock.calls.length).toBe(0);

    wrapper.unmount();

    expect(spyWillUnmount.mock.calls.length).toBe(1);
  });

  it("should call 'getImg' method during componentDidUpdate", async () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();

    const spyDidUpdate = jest.spyOn(instance, 'componentDidUpdate');
    const spyHandleImgShow = jest.spyOn(instance, 'handleImgShow');

    wrapper.setProps({
      ...wrapper.props(),
      result: {
        ...result,
        id: 2,
        image: '',
      },
    });

    expect(spyDidUpdate.mock.calls.length).toBe(1);

    expect(spyHandleImgShow).toHaveBeenCalledTimes(1);

    await expect(getImg).toBeCalled();

    expect(instance.state.imgUrl).toBe(BASE64_IMG_JPEG + img);
  });

  it("should render 'imgUrl' right", () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();

    const spyHandleImgShow = jest.spyOn(instance, 'handleImgShow');

    wrapper.setProps({
      ...wrapper.props(),
      result: {
        ...result,
        id: 2,
        image: `${BASE64_IMG_JPEG}${img}`,
      },
    });

    expect(spyHandleImgShow).toHaveBeenCalledTimes(1);

    expect(instance.state.imgUrl).toBe(BASE64_IMG_JPEG + img);
  });

  it("should call 'handleImgClick' when Avatar clicked", () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();

    const avatar = wrapper.find('Avatar');

    const spyHandleImgClick = jest.spyOn(instance, 'handleImgClick');

    avatar.simulate('click');

    expect(spyHandleImgClick).toBeCalled();
  });

  it("should call 'handleMouseDown' and 'onDelete' when div content clicked and event args is 'resultDel'", () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();

    const divhand = wrapper.find('.divhand');

    const spyHandleMouseDown = jest.spyOn(instance, 'handleMouseDown');

    divhand.simulate('mouseDown', { target: { name: 'resultDel' } });

    expect(spyHandleMouseDown).toBeCalled();
  });

  it("should call 'handleMouseDown' and 'onSelect' when div content clicked and event args is 'resultSelect'", () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();

    const divhand = wrapper.find('.divhand');

    const spyHandleMouseDown = jest.spyOn(instance, 'handleMouseDown');

    divhand.simulate('mouseDown', { target: { name: 'resultSelect' } });

    expect(spyHandleMouseDown).toBeCalled();
  });
});
