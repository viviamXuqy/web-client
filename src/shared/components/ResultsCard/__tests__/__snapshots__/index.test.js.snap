// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsCard component should render right thing and don't change unexpected 1`] = `
<div>
  <div
    className="results-card-div"
  >
    <div
      className="ant-spin-nested-loading"
      style={null}
    >
      <div>
        <div
          className="ant-spin ant-spin-spinning"
        >
          <span
            className="ant-spin-dot ant-spin-dot-spin"
          >
            <i />
            <i />
            <i />
            <i />
          </span>
        </div>
      </div>
      <div
        className="ant-spin-container ant-spin-blur"
      >
        <span
          className="ant-avatar ant-avatar-lg ant-avatar-square"
          onClick={[Function]}
        >
          <span
            className="ant-avatar-string"
          />
        </span>
      </div>
    </div>
    <div
      className="results-card-content"
    >
      <div
        className="divhand"
        onMouseDown={[Function]}
      >
        <p>
          <span
            className="results-card-label"
          >
            <span>
              结果ID
            </span>
            : 
          </span>
          <span
            className="results-card-label-content"
          >
            1
          </span>
        </p>
        <p>
          <span
            className="results-card-label"
          >
            <span>
              任务ID
            </span>
            : 
          </span>
          <span
            className="results-card-label-content"
          >
            1
          </span>
        </p>
        <p>
          <span
            className="results-card-label"
          >
            <span>
              时间
            </span>
            : 
          </span>
          <span
            className="results-card-label-content"
          >
            2018-06-19 05:06:06
          </span>
        </p>
        <p>
          <span
            className="results-card-label"
          >
            <span>
              类型
            </span>
            : 
          </span>
          <span
            className="results-card-label-content"
          >
            car
          </span>
        </p>
        <p>
          <span
            className="results-card-label"
          >
            <span>
              任务类型
            </span>
            : 
          </span>
          <span
            className="results-card-label-content"
          >
            <span>
              实时监控
            </span>
          </span>
          <span
            className="results-card-content__btnGroups"
          >
            <button
              className="ant-btn ant-btn-dashed ant-btn-sm"
              disabled={false}
              name="resultDel"
              onClick={[Function]}
              type="button"
            >
              <i
                className="anticon anticon-delete large-icon"
              />
            </button>
            <label
              className="ant-radio-wrapper"
              onMouseEnter={undefined}
              onMouseLeave={undefined}
              style={undefined}
            >
              <span
                className="ant-radio"
                style={Object {}}
              >
                <input
                  autoFocus={undefined}
                  checked={false}
                  className="ant-radio-input"
                  disabled={undefined}
                  id={undefined}
                  name="resultSelect"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  readOnly={undefined}
                  tabIndex={undefined}
                  type="radio"
                  value={undefined}
                />
                <span
                  className="ant-radio-inner"
                />
              </span>
            </label>
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
